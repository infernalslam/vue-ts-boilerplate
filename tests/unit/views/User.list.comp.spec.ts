import { shallowMount } from '@vue/test-utils';
import UserList from '@/views/User/List.vue';

describe('User : List.vue', () => {
  it('init component properly', () => {
    const wrapper = shallowMount(UserList);

    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.contains('ul')).toBe(true);

    wrapper.vm.$nextTick(() => {
      expect(wrapper.findAll('li')).toHaveLength(4);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
