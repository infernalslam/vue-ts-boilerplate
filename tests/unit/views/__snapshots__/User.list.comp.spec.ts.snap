// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User : List.vue init component properly 1`] = `
<div
  class="user-list"
>
  <h3>
    User List
  </h3>
   
  <ul>
    <li>
      Wisarut Phuv
    </li>
    <li>
      Jack Ma
    </li>
    <li>
      Steve Jobs
    </li>
    <li>
      Nati 
    </li>
  </ul>
</div>
`;
